src/client/connection.ts

```diff
@@ -57,10 +55,9 @@ export const INTENTIONAL_DISCONNECT_CODE = 4000
 type WebsocketState = 0 | 1 | 2 | 3
 
 function getAgent(url: string, config: ConnectionOptions): Agent | undefined {
-  // TODO: replace deprecated method
   if (config.proxy != null) {
-    const parsedURL = parseURL(url)
-    const parsedProxyURL = parseURL(config.proxy)
+    const parsedURL = new URL(url)
```

Comment:
Uses the built in `URL`

Reply:
You have to specify each `key` in `proxyOptions`, because w/ the `URL` class `{...URL}` doesn't give you the values of getters.

Reply:
Should this be a comment in the code for context?

Reply:
Nah, I think its okay without it.